### ISSUE TITLE
User is being prompt to interact when using 'minishift delete --force' command

### ISSUE THREAD

**Issue by odockal** [minishift/minishift#1386]
As a follow up to 'let's create an issue' from https://github.com/minishift/minishift/issues/1320, it has finally happened that I am not able to delete registered rhel image without being asked to interact to minishift's prompt. To be exact was is happening:

`[odockal@odockal-pc latest-nightly]$ ./minishift version`
`minishift v1.5.0+f5e9a94`
`CDK v3.0.0-13092017-1`
`[odockal@odockal-pc latest-nightly]$ ./minishift status`
`Stopped`
`[odockal@odockal-pc latest-nightly]$ ./minishift delete --force`
`Current Minishift VM is registered, but unregistration failed. Do you want to continue [y/N]?:`

From the mentioned discussion in the issue #1320 I did understand that if I am using --force when deleting the VM, there will be no prompts that the user must react to, and minishift will deal with all such situations - deleting vm-specific files even if vm was not stopped and/or unregistration failed (when using image that requires registration, like is happening in cdk).

Unfortunately I am not able to give a steps to reproduce this. But I can try to find out something.

**Comment by gbraad** [comment]
Current version of Minishift does not ask for confirmation, and has no force option. CDK has?

In the current implementation, the VM needs to be running to unregister. Will check if the message is correctly returned.

**Comment by gbraad** [comment]
I do remember the registration state was a pinpoint in some of the discussion. To not have the user end up in a situation in which he needs to take extra steps to reclaim the subscription we disallowed the deletion when registered. Can you try to use --skip-unregistration

For a force, do you expect this to be implied?

**Comment by gbraad** [comment]
In the current implementation we do not allow `--skip-unregistration` for `m delete`:

https://github.com/minishift/minishift/blob/8aed58a3f36403958f4a2ad4bb95e5a241316844/cmd/minishift/cmd/delete.go#L53

however, we allow this to be used for `m stop`: 

https://github.com/minishift/minishift/blob/8aed58a3f36403958f4a2ad4bb95e5a241316844/cmd/minishift/cmd/stop.go#L59

**Comment by odockal** [comment]
@gbraad I do recall that I used `m stop --skip-unregistration` while trying to clear minishift/cdk state, but that just seems to me like workaround. Let's see what is opinion of others. Thanks for updating.

**Comment by LalatenduMohanty** [comment]
This need some code refactor as we ignore the --force in the handleFailedRegistrationAction() function
```
func UnregisterHost(api libmachine.API, allowToSkipUnregistration bool) {
	if allowToSkipUnregistration && SkipUnRegistration {
		log.Debug("Skipping unregistration due to enabled '--skip-unregistration' flag")
		return
	}

	if !isRegistered() {
		return
	}

	if wasSuccessful, err := cluster.UnRegister(api); err != nil || !wasSuccessful {
		handleFailedRegistrationAction()
	}

	// if continued, we remove the IsRegistered state
	setRegistered(false)
}

func handleFailedRegistrationAction() {
	hasConfirmed := pkgUtil.AskForConfirmation("Current Minishift VM is registered, but unregistration failed.")
	if !hasConfirmed {
		atexit.ExitWithMessage(0, fmt.Sprintln("Aborted."))
	}
}
```

**Comment by praveenkumar** [comment]
@odockal We have this because of below reason. I don't think we should take any action around it if the VM is not responsive.

https://github.com/minishift/minishift/blob/master/cmd/minishift/cmd/registration/util.go#L68 here we are performing and unregistration first and if that fails then only prompt for this message.

- We only get single subscription when sign up using developer program (developer.redhat.com) and there is now way we should force user to delete a VM which already subscribed.

- We do have --skip-registration which work on start command and you can spin RHEL vm without registration and then when you delete you will not prompt for that because vm is not subscribed.

**Comment by LalatenduMohanty** [comment]
Ideally we should respect `--force`. Even if it can bring disastrous results. Otherwise we are breaks the general perception of the meaning of `--force` flag. 
However  we should figure out how difficult it is for a developer (with a developer subscription) to free up the entitlement if user has deleted the VM without un-registering it. If it is really difficult than we might want to differ this issue.

**Comment by odockal** [comment]
This is definitely worth exploring, what happens if vm is deleted and image was not unsubscribed. I would not vote for prioritizing ease of automation over developer's experience or troubles having multiple subscribed images over developer.redhat.com. 

I am not sure how this works as I am having multiple images subscribed under my test account and I did not have any problem like exceeding limit of number of subscriptions. Although, I did not do any thorough analysis around number of possible subscribed image (via cdk).
